engine/pkg/logutil/log_test.go

```diff
@@ -25,42 +29,61 @@ import (
 	"go.uber.org/zap"
 )
 
-var testFile = "test.log"
-
-func init() {
-	logger, prop, err := log.InitLogger(&log.Config{
+func TestNewLogger(t *testing.T) {
+	testFile := filepath.Join(t.TempDir(), "test.log")
+	lg, prop, err := log.InitLogger(&log.Config{
 		Level: "warn",
 		File: log.FileLogConfig{
 			Filename: testFile,
 		},
 	})
-	if err != nil {
-		panic(err)
-	}
-	log.ReplaceGlobals(logger, prop)
-}
+	require.NoError(t, err)
+	log.ReplaceGlobals(lg, prop)
 
-func TestNewLogger(t *testing.T) {
-	defer os.Remove(testFile)
-	tail, err := tail.TailFile(testFile, tail.Config{Follow: true})
+	var wg sync.WaitGroup
+	defer wg.Wait()
+
+	tailHandler, err := tail.TailFile(testFile, tail.Config{Follow: true, Poll: true})
 	require.NoError(t, err)
-	defer tail.Stop()
+	defer tailHandler.Stop()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
```

Comment:
Too short a timeout may also lead to unstable tests, especially in a slow CI environment. If we can ensure test logic is correct and can pass fast, I think it's ci's responsibility to avoid the test running so long.

Reply:
Make it fast fail.